defaults/api_ents.go

```diff
@@ -0,0 +1,112 @@
+package defaults
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/moby/libentitlement/entitlement"
+	"github.com/moby/libentitlement/secprofile"
+)
+
+const (
+	// APIEntFullID is the API entitlement identifier; the value format is: "api.access:api-id:subset:[allow|deny]"
+	// ex: "api.access:engine.swarm:all:allow"
+	APIEntFullID = "api.access"
+
+	// APIFullControl specifies access control for the whole api
+	APIFullControl = "all"
+)
+
+// Default known APIs and API subsets to control access of
+const (
+	// EngineAPI defines the Moby-Engine API
+	EngineAPI = "engine"
+
+	// SwarmAPI defines the Moby-Engine Swarm API
+	SwarmAPI = "swarm"
+)
+
+var (
+	apiEntitlement = entitlement.NewStringEntitlement(APIEntFullID, apiEntitlementEnforce)
+)
+
+// GetSwarmAPIIdentifier returns the full Swarm API identifier
+func GetSwarmAPIIdentifier() secprofile.APIID {
+	return secprofile.APIID(fmt.Sprintf("%s.%s", EngineAPI, SwarmAPI))
+}
+
+/*IsSwarmAPIControlled checks if Moby Swarm API is controlled and whether it's allowed or not
+ * Return values are the following:
+ * isControlled - if no error is encountered, whether the Swarm API is currently controlled by the entitlements
+ * access - if Swarm API is currently controlled, this return value holds the allow/deny access requested
+ * err - error returned if an issue is encountered
+ */
+func IsSwarmAPIControlled(profile secprofile.Profile) (isControlled bool, access secprofile.APIAccess, err error) {
+	ociProfile, err := ociProfileConversionCheck(profile, APIEntFullID)
+	if err != nil {
+		return false, secprofile.Allow, err
+	}
+
+	if ociProfile.APIAccessConfig == nil {
+		return false, secprofile.Allow, fmt.Errorf("OCI profile's APIAccess field nil")
+	}
+
+	swarmAPIFullID := GetSwarmAPIIdentifier()
+
+	apiSubset, ok := ociProfile.APIAccessConfig.APIRights[swarmAPIFullID]
+	if !ok || apiSubset == nil {
+		return false, secprofile.Allow, nil
+	}
+
+	access, ok = apiSubset[APIFullControl]
+	if !ok {
+		return false, secprofile.Allow, nil
+	}
+
+	return true, access, nil
+}
+
+func apiEntitlementEnforce(profile secprofile.Profile, apiToAccess string) (secprofile.Profile, error) {
+	logrus.Debugf("API entitlement for %s", apiToAccess)
+
+	ociProfile, err := ociProfileConversionCheck(profile, APIEntFullID)
+	if err != nil {
+		return nil, err
+	}
+
+	if ociProfile.APIAccessConfig == nil {
+		return nil, fmt.Errorf("OCI profile's APIAccess field nil")
+	}
+
+	apiToAccessFields := strings.Split(apiToAccess, ":")
+	badAccessFormatError := fmt.Errorf("Wrong API subset and access format, should be \"api-id:subset:[allow|deny]\"")
+
+	logrus.Debugf("Fields found for %s: %v", apiToAccess, apiToAccessFields)
+	if len(apiToAccessFields) != 3 {
+		return nil, badAccessFormatError
+	}
+
+	apiIDStr := apiToAccessFields[0]
+	apiSubsetStr := apiToAccessFields[1]
```

Comment:
Is the subset string the raw API path or a different mapping?  And if so to the latter, how is it specified?

And do we plan to limit certain arguments?  For example, `--privileged` since it's [settable with container/create](https://docs.docker.com/engine/api/v1.33/#operation/ContainerCreate)?